frontend/src/App.tsx

```diff
@@ -1,11 +1,63 @@
 import "./App.css";
+import { BrowserRouter, Routes, Route } from "react-router-dom";
+import SideBar from "./components/SideBar/SideBar";
+import Feed from "./screens/Feed/Feed";
+import History from "./screens/History/History";
+import Dashboard from "./screens/Dashboard/Dashboard";
+import Login from "./screens/auth/Login/Login";
+import SignUp from "./screens/auth/SignUp/SignUp";
+import { Navigate } from "react-router-dom";
 
-function App() {
+type ProtectedRouteProps = {
+  children: React.ReactNode;
+};
+
+const ProtectedRoute: React.FC<ProtectedRouteProps> = ({ children }) => {
```

Comment:
nice! does this chunk of code redirect users to login if there's no token?

Reply:
Yes, it redirects you back to the Login page if the token has not been set.

Reply:
really cool! great job handling this edge case